src/components/forms/form.tsx

```diff
@@ -0,0 +1,27 @@
+import {
+  Form as FormPF4,
+  FormProps as FormPF4Props,
+} from '@patternfly/react-core';
+import React from 'react';
+import { Omit } from 'react-redux';
+
+type FormProps = Omit<FormPF4Props, 'onSubmit'>;
+
+const ignoreSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  event.preventDefault();
+  const submitButton = document.querySelector(
+    'button.pf-m-primary[type="submit"]'
+  ) as HTMLButtonElement;
+  submitButton.click();
```

Comment:
This makes the assumption that every form will have a button that matches the class 'button.pf-m-primary[type="submit"]'. Instead of making an assumption that button exists, it would be preferable to have the button's onClick call `preventDefault`.

For example, in ocpDetails/detailsTag.tsx, I do this in the button's onClick function.

```
public handleOpen = event => {
    this.setState({ isOpen: true });
    event.preventDefault();
    return false;
  };
```

And in azureDataSource/index.tsx, azureCreds/index.tsx, etc. I see that you have been doing this:

```
<Form
      onSubmit={e => {
        e.preventDefault();
        return false;
      }}
    >
```

Reply:
I am not sure I understood what you were saying.
If I'm not wrong when you're pressing Enter in the input box, it triggers the "submit" event.
Although I agree with you that I'm doing here a bad assumption, I don't think calling preventDefault from the onClick button will fix it. Did you try it?

Reply:
In addition, the requirement was to click on the Next button on pressing enter on the input box so I'm not sure how that works.

Reply:
Intresting....

Reply:
Unfortunately, in the wizard I can not set the "onClick" props of the buttons.
Nevertheless, even if it was possible, the problem was pressing enter when you're on the input box. But going to every text input and adding `onClick={e => e.preventDefault()}` doesn't sound so good.

I do understand that I'm doing an assumption of having the submit button. So what I can do is move this to `createCostModelWizard` instead putting it in `components`

Reply:
Pressing enter in the input box shouldn’t navigate the user to the next step. (Perhaps set focus on the button?) The user shouldn’t navigate elsewhere unless they click on the next button. Is that what’s happening?

Reply:
there is a conflict then. @nlcwong - in the fit & finish I remembered you mentioning on pressing enter in the input box was to move to the next button

![image](https://user-images.githubusercontent.com/2453279/69188753-caa92680-0b25-11ea-8235-ed3201dd6a16.png)

Reply:
Looking at the PatternFly wizard example and the text input does not navigate the user to the next step. Although, that could be a PatternFly issue?

Reply:
What i wrote on the box is the issue.  It's not supposed to behave that way.  When a user clicks the "Enter" key, it should not auto-advance the step UNLESS the focus is on the primary button.  If the focus is on the primary button, wouldn't any key then make the button depress and do that action?

Reply:
OK thanks.